qiskit/circuit/quantumcircuit.py

```diff
@@ -1496,8 +1500,8 @@ def decompose(
         from qiskit.converters.circuit_to_dag import circuit_to_dag
         from qiskit.converters.dag_to_circuit import dag_to_circuit
 
-        pass_ = Decompose(gates_to_decompose=gates_to_decompose)
-        decomposed_dag = pass_.run(circuit_to_dag(self))
+        pass_ = Decompose(gates_to_decompose=gates_to_decompose, reps=reps)
+        decomposed_dag = pass_.run(dag=circuit_to_dag(self))
```

Comment:
I think the previous strategy in this PR of having the repeated run here was cleaner.  There, `reps` was clearly a number of repeats of the whole pass - with it mixed in to the pass itself, it's much less clear (and the `Decompose` pass is pretty much only used by this function).